near-contract-standards/src/non_fungible_token/core/core_impl.rs

```diff
@@ -280,6 +280,65 @@ impl NonFungibleToken {
         // return previous owner & approvals
         (owner_id, approved_account_ids)
     }
+
+    /// Mint a new token. Not part of official standard, but needed in most situations.
+    /// Consuming contract expected to wrap this with an `nft_mint` function.
+    ///
+    /// Requirements:
+    /// * Caller must be the `owner_id` set during contract initialization.
+    /// * Caller of the method must attach a deposit of 1 yoctoⓃ for security purposes.
+    /// * If contract is using Metadata extension (by having provided `metadata_prefix` during
+    ///   contract initialization), `token_metadata` must be given.
+    /// * token_id must be unique
+    ///
+    /// Returns the newly minted token
+    pub fn mint(
```

Comment:
still need to mark it as payable i think

Reply:
actually, this is doing the opposite, which is not exposing this function for external calls at all. 

Looking at this again, I think this function can just be removed altogether, the mint functionality can still be called through the `NonFungibleToken` as the [NFT example does](https://github.com/near/near-sdk-rs/blob/cf3ef24ee51449b48a17246a5744a714e13f660b/examples/non-fungible-token/nft/src/lib.rs#L98). I was just trying to mitigate breaking changes but I think this just causes more issues for no benefit.

Any objections from removing this function from the NFT trait which is [not part of the standard](https://nomicon.io/Standards/NonFungibleToken/Core.html) and this no longer by default exposes the `mint` function which is currently broken anyway on its current version? cc @mikedotexe

Reply:
got your point.
still prefer to keep this function, mark it as payable to prevent any breaking changes. But if you guys are ok to remove this function totally I'm good with it.

Reply:
> got your point.
> still prefer to keep this function, mark it as payable (other mechanism shall be used to make it internal), in order to avoid any breaking changes. But if you guys are ok to remove this function totally I'm good with it.

well, the only breaking change is no longer externally exposing a function that was broken and should not have been included in the standard in the first place. You can still use the mint functionality as the example does, too. I'll wait on @mikedotexe's take or maybe @chadoh since he worked on the standard also.